### ISSUE TITLE
False positive: `no-unreachable` unexpectedly flags statements inside catch clause

### ISSUE THREAD

**Issue by magurotuna** [denoland/deno_lint#716]
### Lint Name

no-unreachable

### Code Snippet

```ts
export function foo(trueOrFalse: boolean) {
  if (trueOrFalse) {
    // noop
  } else {
    // noop
  }
  try {
    bar();
    return 42;
  } catch (err) {
    console.error(err);
  }
}
```

### Expected Result

No error happens

### Actual Result

```
(no-unreachable) This statement is unreachable
    console.error(err);
    ^^^^^^^^^^^^^^^^^^^
    at /home/yusuktan/Repos/github.com/magurotuna/deno_lint/maguro.ts:11:4

Found 1 problem
Checked 1 file
```

### Additional Info

If the if-else block is removed this false positive goes away.

I found this bug when browsing oak's code
https://github.com/oakserver/oak/blob/51c02fc1979c8e2f921f91099a390ecc4727fb7a/application.ts#L428-L433

### Version

deno 1.10.3 (release, x86_64-unknown-linux-gnu)
v8 9.1.269.27
typescript 4.2.2
